### PR TITLE
feat(cli): add namespace column to vcctl pod list --all-namespaces

### LINKED ISSUE #5647: vcctl pod list --all-namespaces output is ambiguous: no Namespace column
### Description

`vcctl pod list --queue <q> --all-namespaces` can print multiple rows that are indistinguishable from one another, because the output table has no `Namespace` column. When two pods share the same name in different namespaces (a common pattern, for example a per-team `worker-0`), the listing shows identical rows and the user cannot tell which namespace each pod belongs to.

This came up while reviewing #5636. That PR fixes a real de-duplication bug: `vcctl pod list --queue` merges vcjob pods with annotation-matched pods and drops duplicates, but it compared pods by name only. With `--all-namespaces` the list spans namespaces, so two distinct pods that happened to share a name in different namespaces were collapsed into one and one of them went missing from the output entirely. #5636 correctly changes the comparison to namespace plus name, so both pods are now listed.

However, once both pods are listed there is still no `Namespace` column, so the two rows look identical and the user still cannot distinguish them. The de-duplication fix and the missing column are two sides of the same usability gap, and the second half remains after #5636.

### Steps to reproduce the issue

1. Create two namespaces and a queue:

   ```yaml
   apiVersion: v1
   kind: Namespace
   metadata:
     name: team-a
   ---
   apiVersion: v1
   kind: Namespace
   metadata:
     name: team-b
   ---
   apiVersion: scheduling.volcano.sh/v1beta1
   kind: Queue
   metadata:
     name: test
   spec:
     weight: 1
     reclaimable: true
   ```

2. Create two pods with the same name in different namespaces, both tied to the queue:

   ```yaml
   apiVersion: v1
   kind: Pod
   metadata:
     name: worker-0
     namespace: team-a
     annotations:
       scheduling.volcano.sh/queue-name: test
   spec:
     containers:
     - name: c
       image: registry.k8s.io/pause:3.9
   ---
   apiVersion: v1
   kind: Pod
   metadata:
     name: worker-0
     namespace: team-b
     annotations:
       scheduling.volcano.sh/queue-name: test
   spec:
     containers:
     - name: c
       image: registry.k8s.io/pause:3.9
   ```

3. Run:

   ```
   vcctl pod list --queue test --all-namespaces
   ```

### Describe the results you received and expected

Before #5636 one of the two pods is silently omitted from the output because same-named pods across namespaces are de-duplicated by name:

```
Name            Ready      Status     Restart  Age
worker-0        0/1        Running    0        28s
```

With the #5636 de-dup fix applied both pods are listed, but the two rows are identical and ambiguous because there is no `Namespace` column:

```
Name            Ready      Status     Restart  Age
worker-0        0/1        Running    0        28s
worker-0        0/1        Running    0        28s
```

Expected: a leading `Namespace` column when `--all-namespaces` is set, mirroring `kubectl get pods --all-namespaces`:

```
Namespace     Name            Ready      Status     Restart  Age
team-a        worker-0        0/1        Running    0        28s
team-b        worker-0        0/1        Running    0        28s
```

### What version of Volcano are you using?

Latest `master` (built from source).

### Any other relevant information

`kubectl get pods -A` always renders a `NAMESPACE` column; `vcctl pod list --all-namespaces` should do the same. A fix is prepared as a PR stacked on top of #5636.

### PR DESCRIPTION
**Pull request by hajnalmt**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

`vcctl pod list --all-namespaces` renders only the pod name, so pods that share a name across namespaces show up as identical, indistinguishable rows. This PR adds a leading `Namespace` column whenever `--all-namespaces` is set, matching the behaviour of `kubectl get pods --all-namespaces`. The default single-namespace output is unchanged.

`PrintPods` is reworked to build its columns dynamically so the optional `Namespace` column can be inserted without duplicating the width and formatting logic.

This PR is stacked on top of #5636. That PR fixes a related bug: `vcctl pod list --queue` compared pods by name only, so with `--all-namespaces` two distinct pods that happened to share a name in different namespaces were de-duplicated by name and one of them was silently omitted from the output. #5636 fixes the omission by comparing namespace and name, and this PR closes the remaining usability gap by making the two rows distinguishable. The first commit here is aeron-gh's unmodified dedup fix; if this PR merges, #5636 will be marked as merged too.

#### Which issue(s) this PR fixes:

Fixes #5647

#### Special notes for your reviewer:

Verification against a live cluster with two same-named pods in different namespaces (`team-a/worker-0` and `team-b/worker-0`), both tied to queue `test`:

```
./_output/bin/vcctl pod list --queue test --all-namespaces
Namespace     Name            Ready      Status                  Restart  Age
team-a        worker-0        0/1        ImagePullBackOff        0        55m
team-b        worker-0        0/1        ImagePullBackOff        0        55m
```

Before this change both rows read `worker-0` with no namespace, so they were indistinguishable. The `ImagePullBackOff` status is unrelated to this change and just reflects the `pause` test image.

Unit tests cover both directions: the `Namespace` column is present with `--all-namespaces` and absent without it.

AI disclosure: this PR was prepared with AI assistance in line with the project AI guidance.

#### Does this PR introduce a user-facing change?

```release-note
`vcctl pod list --all-namespaces` now shows a `Namespace` column so pods with the same name in different namespaces are distinguishable.
```

### CODE REVIEW

File: pkg/cli/pod/pod.go [hunk 1]
```diff
@@ -150,54 +153,72 @@ func ListPods(ctx context.Context) error {
...
+	const columnSpacing = 8
+	widths := make([]int, len(columns))
+	for i, col := range columns {
+		for _, info := range infoList {
+			if l := len(col.value(info)); l > widths[i] {
+				widths[i] = l
+			}
 		}
+		widths[i] += columnSpacing
 	}
```
[the remark this answers is not in the document]
Comment (hajnalmt) [reply pkg/cli/pod/pod.go:200]: Okay, but this is not this easy.
widths[i] = len(header) then += spacing will widen every column, so I decided to go with `max(headerLen, maxValueLen + spacing)` but this is fixed now.

[the remark this answers is not in the document]
Comment (hajnalmt) [reply pkg/cli/pod/pod.go:200]: max(headerLen, maxValueLen + spacing) was used in the end to not widen every column thanks.

File: pkg/cli/pod/pod.go [hunk 2]
```diff
@@ -150,54 +153,67 @@ func ListPods(ctx context.Context) error {
...
+	const columnSpacing = 8
+	widths := make([]int, len(columns))
+	for i, col := range columns {
+		widths[i] = len(col.header)
+		for _, info := range infoList {
+			if l := len(col.value(info)) + columnSpacing; l > widths[i] {
+				widths[i] = l
+			}
 		}
 	}
```
[the remark this answers is not in the document]
Comment (hajnalmt) [reply pkg/cli/pod/pod.go:200]: This is such a pathological edge case. But added with a `headerSpacing=2`, a single space must be scarce if we care about this. I also added a unit test too, column width is now `max(headerLen + 2, maxValueLen + 8)`.

File: pkg/cli/pod/pod.go [hunk 3]
```diff
@@ -150,54 +153,67 @@ func ListPods(ctx context.Context) error {
...
+	formatStr := strings.Join(formatBuilder, "") + "\n"
+
+	headerArgs := make([]interface{}, len(columns))
+	for i, col := range columns {
+		headerArgs[i] = col.header
+	}
+	if _, err := fmt.Fprintf(writer, formatStr, headerArgs...); err != nil {
 		fmt.Printf("Failed to print Pod information: %s.\n", err)
 		return
 	}
```
[the remark this answers is not in the document]
Comment (hajnalmt) [reply pkg/cli/pod/pod.go:215]: Pre-existing behavior: the `fmt.Printf(...)` on write failure was already there before this PR. Also this "issue" is there at a lot of places. Let's keep the PR focused.

File: pkg/cli/pod/pod.go [hunk 4]
```diff
@@ -150,54 +153,67 @@ func ListPods(ctx context.Context) error {
...
-		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
-		if err != nil {
+		rowArgs := make([]interface{}, len(columns))
+		for i, col := range columns {
+			rowArgs[i] = col.value(info)
+		}
+		if _, err := fmt.Fprintf(writer, formatStr, rowArgs...); err != nil {
 			fmt.Printf("Failed to print Pod information: %s.\n", err)
 			return
 		}
```
[the remark this answers is not in the document]
Comment (hajnalmt) [reply pkg/cli/pod/pod.go:224]: Pre-existing behavior: the `fmt.Printf(...)` on write failure was already there before this PR. Also this "issue" is there at a lot of places. Let's keep the PR focused.

**Comment by aeron-gh** [comment]
One small thought
feel free to ignore it if you'd rather keep it for a follow-up. I noticed that in the --queue path (which merges two filtered lists) the rows aren't grouped by namespace under `--all-namespaces,` unlike the single-list paths that come back in the API's namespace/name order. An explicit sort by (namespace, name) before printing would keep it grouped.
If you think that makes sense, I'm happy to send a small follow-up PR for it.

**Comment by hajnalmt** [comment]
> One small thought feel free to ignore it if you'd rather keep it for a follow-up. I noticed that in the --queue path (which merges two filtered lists) the rows aren't grouped by namespace under `--all-namespaces,` unlike the single-list paths that come back in the API's namespace/name order. An explicit sort by (namespace, name) before printing would keep it grouped. If you think that makes sense, I'm happy to send a small follow-up PR for it.

Sure, great idea go ahead and prepare the PR @aeron-gh ! 👍
Thanks for the merge Jesse ^^
